src/javis_viewer.jl

```diff
@@ -0,0 +1,292 @@
+        if frames > curr_frame
+            push!(slide, curr_frame + 1)
+            @guarded draw(canvas) do widget
+                # Gets the next Javis frame based on textbox entry
+                frame_mat = transpose(get_javis_frame(video, action_list, curr_frame + 1))
+                # Gets the correct Canvas context to draw on
+                context = getgc(canvas)
+                image(
+                    context,
+                    CairoImageSurface(frame_mat),
+                    0,
+                    0,
+                    frame_dims[1],
+                    frame_dims[2],
+                )
+            end
+        else
+            push!(tbox, 1) # Sets the first frame shown to one
+            @guarded draw(canvas) do widget
+                # Gets the next Javis frame based on textbox entry
+                frame_mat = transpose(get_javis_frame(video, action_list, 1))
+                # Gets the correct Canvas context to draw on
+                context = getgc(canvas)
+                image(
+                    context,
+                    CairoImageSurface(frame_mat),
+                    0,
+                    0,
+                    frame_dims[1],
+                    frame_dims[2],
+                )
+            end
+        end
+    end
+
+    # When the `Right Arrow` key is pressed, increment current frame number
+    signal_connect(win, "key-press-event") do widget, event
+        if event.keyval == 65363
+            curr_frame = parse(Int, get_gtk_property(tbox, :text, String))
+            if frames > curr_frame
+                push!(slide, curr_frame + 1)
+                @guarded draw(canvas) do widget
+                    # Gets the next Javis frame based on textbox entry
+                    frame_mat =
+                        transpose(get_javis_frame(video, action_list, curr_frame + 1))
+                    # Gets the correct Canvas context to draw on
+                    context = getgc(canvas)
+                    image(
+                        context,
+                        CairoImageSurface(frame_mat),
+                        0,
+                        0,
+                        frame_dims[1],
+                        frame_dims[2],
+                    )
+                end
+            else
+                push!(tbox, 1) # Sets the first frame shown to one
+                @guarded draw(canvas) do widget
+                    # Gets the next Javis frame based on textbox entry
+                    frame_mat = transpose(get_javis_frame(video, action_list, 1))
+                    # Gets the correct Canvas context to draw on
+                    context = getgc(canvas)
+                    image(
+                        context,
+                        CairoImageSurface(frame_mat),
+                        0,
+                        0,
+                        frame_dims[1],
+                        frame_dims[2],
+                    )
```

Comment:
this looks like it should go into a method as you use this more than once.

Reply:
Probably put the whole `@guarded` one into a method and call with the `video, action_list, frame_number`

Reply:
Good point. I fixed this up. Let me know if you think I managed it properly.